### ISSUE TITLE
pytorch to .onnx missing operator

### ISSUE THREAD

**Issue by shlomi-amitai** [pytorch/pytorch#46586]
## 🚀 Feature

torch.ROLL
## Motivation
want to shift an array

gets the error :
RuntimeError: Exporting the operator roll to ONNX opset version 11 is not supported. Please open a bug to request ONNX export support for the missing operator.

## Additional context
pytorch version 18.5

**Comment by malfet** [comment]
Can you please run `python3 -mtorch.utils.collect_env` and paste its output here?

**Comment by spandantiwari** [comment]
@shlomi-amitai - is there a model where this op is needed? 
We don't have this on our immediate roadmap. Please consider submitting a PR for this.

**Comment by MohammadMahdiKazemi** [comment]
I encountered this issue converting an action recognition model where a time-shift is needed.

**Comment by knaffe** [comment]
Now the [Swin Transformer](https://github.com/microsoft/Swin-Transformer) is needed this roll OP for ONNX.
`RuntimeError: Exporting the operator roll to ONNX opset version 10 is not supported. Please open a bug to request ONNX export support for the missing operator.`

**Comment by XinWangg** [comment]
@knaffe Have you solve the problem? I am converting swin transformer too..

**Comment by serser** [comment]
@knaffe @XinWangg  Please checkout [Twins](https://github.com/Meituan-AutoML/Twins) that is roll-op free and onnx-exportable.

**Comment by qiaoyu-debug** [comment]
@knaffe You can use operator_export_type=OperatorExportTypes.ONNX_FALLTHROUGH to export onnx, then the model contains an undefined op "roll". Last, parser onnx and write trt plugin

**Comment by qiaoyu-debug** [comment]
@XinWangg  You can use operator_export_type=OperatorExportTypes.ONNX_FALLTHROUGH to export onnx, then the model contains an undefined op "roll". Last, parser onnx and write trt plugin

**Comment by BowenBao** [comment]
Should be supported by #58697

**Comment by MASON1211** [comment]
@serser Hi I wanted to use Swin Transformer in some production environment. Are you sure that Twins is onnx-exportable and have you verified it ? Just need to be sure..

**Comment by serser** [comment]
> @serser Hi I wanted to use Swin Transformer in some production environment. Are you sure that Twins is onnx-exportable and have you verified it ? Just need to be sure..

Yes, I've exported twins into onnx without problem. In case you'd like TensorRT deployment for downstream tasks, you may need to cope with dynamic inputs, just fix the input will do.

**Comment by AleiChan** [comment]
@serser Hi, i want to use Swin Transformer to onnx, but the operator roll is unwork, so why Twins is ok, and i have  trained the model of Swin Transformer ,  i want to know can i deal it.

**Comment by AleiChan** [comment]
> Now the [Swin Transformer](https://github.com/microsoft/Swin-Transformer) is needed this roll OP for ONNX.
> `RuntimeError: Exporting the operator roll to ONNX opset version 10 is not supported. Please open a bug to request ONNX export support for the missing operator.`

hi, i wanna to know have you solved this problem？ can you help me , please~ @knaffe

**Comment by MASON1211** [comment]
@serser oh I am sorry this is for image classification , I am using Swin Transformer for object detection task.

**Comment by MASON1211** [comment]
> Should be supported by #58697

I am still facing this issue , using onnx '1.9.0'

**Comment by ma-siddiqui** [comment]
Hi,

Any resolution on this issue?

Thanks,

**Comment by BowenBao** [comment]
@MASON1211 @ma-siddiqui #58697 was merged in master recently which will appear in the next PyTorch release. You can try the PyTorch nightly version for now.

**Comment by serser** [comment]
> @serser oh I am sorry this is for image classification , I am using Swin Transformer for object detection task.

Twins support OD, Seg as well.

**Comment by serser** [comment]
> @serser Hi, i want to use Swin Transformer to onnx, but the operator roll is unwork, so why Twins is ok, and i have trained the model of Swin Transformer , i want to know can i deal it.

@AleiChan Twins doesn't have `roll` operation.

**Comment by serser** [comment]
> @MASON1211 @ma-siddiqui #58697 was merged in master recently which will appear in the next PyTorch release. You can try the PyTorch nightly version for now.

@BowenBao Great job! Have you tried converting such-exported onnx model to TensorRT engine?

**Comment by jili5ms** [comment]
+1. Supporting roll op would be great as swin is very popular now.

**Comment by troylhy1991** [comment]
HI, I have a dirty fix temporarily, you can try to replace torch.roll with similar roll function mentioned  [here](https://discuss.pytorch.org/t/implementation-of-function-like-numpy-roll/964/10), and swin-transformer could be exported to onnx successfully; however, it might not be very efficient. Other methods could be implementing custom_roll with cpp/cuda and register the op or just wait for onnx's update ;)

**Comment by dashesy** [comment]
Any plans for adding support of `torch.roll` to ONNX?

**Comment by dashesy** [comment]
Yes PyTorch 1.10 has the ops, thanks
